src/Symfony/Component/ExpressionLanguage/Node/BinaryNode.php

```diff
@@ -173,13 +173,13 @@ public function toArray()
         return ['(', $this->nodes['left'], ' '.$this->attributes['operator'].' ', $this->nodes['right'], ')'];
     }
 
-    private function evaluateMatches(string $regexp, string $str): int
+    private function evaluateMatches(string $regexp, string $str = null): int
```

Comment:
I'd prefer writing this as `?string $str` since the argument is not optional.
Also, we should do a similar cast to string on L55 I think.

Reply:
I wasn’t sure if `?string` is already supported by the lowest PHP version for 4.4 ☺️

Updated in 30e2382cf17826e569b300064178e9021f7a02d8 and 399c3e3b45e9ff3c123801c0fd3d3a3679a3be52